src/tutorial/furthertopics/message-triggers.rst

```diff
@@ -0,0 +1,433 @@
+            echo $newrand >> report.txt;
+            counter=$[($counter + 1)];
+         done
+
+
+   #. **Create a new suite.**
+
+      Create a ``suite.rc`` file and paste the following basic suite into it:
+
+      .. code-block:: cylc
+
+         [cylc]
+             UTC mode = True
+
+         [meta]
+             title = "test suite to demo message triggers"
+
+         [scheduling]
+             initial cycle point = 2019-06-27T00Z
+             final cycle point = 2019-10-27T00Z
+                
+             [[dependencies]]
+
+                 [[[P2M]]]
+                     graph = """
+                         long_forecasting_task =>  another_weather_task
+                         long_forecasting_task => different_weather_task 
+                         long_forecasting_task[-P2M] => long_forecasting_task
+                     """
+
+      This is a basic suite, currently it does not have any message triggers
+      attached to any task.
+
+
+   #. **Define our tasks in the runtime section.**
+
+      Next we want to create our ``runtime`` section of our suite.
+      First we define what the tasks do. In this example 
+      ``long_forecasting_task`` will sleep, create a file containing some
+      random numbers and produce a message.
+      (Note that the random number generator bash script has already been 
+      preloaded into your ``bin`` directory.)
+      ``another_weather_task`` and ``different_weather_task`` simply sleep.
+
+      Add the following code to the  ``suite.rc`` file. 
+
+      .. code-block:: cylc
+
+         [runtime]
+
+             [[long_forecasting_task]]
+                 script = """
+                         sleep 2
+                         random.sh
+                        
+                         sleep 2
+                         random.sh
+                                            
+                         sleep 2
+                         random.sh
+                     """
+
+             [[another_weather_task, different_weather_task]]
+                 script = sleep 1
+
+      
+   #. **Create message triggers.**
+
+      We now have a suite with a task, ``long_forecasting_task`` which, after
+      it has fully completed, triggers two more tasks, ``another_weather_task``
+      and ``different_weather_task``.
+
+      Suppose we want ``another_weather_task`` and ``different_weather_task``
+      to start before ``long_forecasting_task`` has fully completed, perhaps
+      after some data has become available. 
+      
+      In this case, we shall trigger ``another_weather_task`` after one set of 
+      random numbers has been created
+      and ``different_weather_task`` after a second set of random numbers has
+      been created.    
+      
+      There are three aspects of creating messsage triggers. 
+      The first is to create the messages within ``runtime`` in our suite
+      - we need to create a sub-section called ``outputs``. Here we create
```

Comment:
This makes it sound like
`[runtime][outputs]` wheras it's actually `[runtime][TASK][outputs]`